build.py

```diff
@@ -285,8 +285,6 @@ def RunCMake(context, extraArgs=None, stages=None):
 def RunCTest(context, extraArgs=None):
     buildDir = context.buildDir
     variant = BuildVariant(context)
-    #TODO we can't currently run tests in parallel, something to revisit.
-    numJobs = 1
```

Comment:
We would like to preserve the default behavior of running tests in serial unless the number of jobs was explicitly passed to the build script. This would match what `ctest` is doing where by default it runs with a single job, but you can enable parallel execution when using `-j` flag. Once this is done, we should be able to merge this PR and start gradually adopt it platform by platform.

Reply:
Good point. I re-worked this change to only use multiple jobs for testing when you explicitly use `--jobs`. Otherwise if --jobs isn't given, the behavior should be the same as before where it uses the number of CPUs for building and only one for testing. Also rebased against dev.

See https://github.com/Autodesk/maya-usd/pull/578/commits/f2cb3513671253f160ed6ff36805ebd26628f86e.

Reply:
Thank you, Matt. I added @HamedSabri-adsk to the review since he may have some feedback for the actual implementation.